docs/type_inference.md

```diff
@@ -66,8 +66,13 @@ a few problems:
 
 The solution is as follows:
 
-1. **For most ops without regions** (like `PadOp`): Put all the verification
-   code into the shape functions, and discard verifiers totally.
+1. **For most ops without regions** (like `PadOp`): Try to put all the verification
+   code into the shape functions, and discard verifiers totally. In cases where
+   this isn't possible due to the inability to infer return types
+   (such as with`ReshapeOp` or `BroadcastInDimOp`), create a verifier to
+   contain the necessary verification logic. Typically inferable ops,
+   like `AddOp`, may still need a verifier to perform additional verifications
+   when quantization is involved.
```

Comment:
"because they are verifying constraints of a provided return type, which is not accessible in the type/shape inference methods."